app/Main.hs

```diff
@@ -1,39 +1,45 @@
 {-# LANGUAGE FlexibleContexts  #-}
 {-# LANGUAGE OverloadedStrings #-}
 
-
 module Main where
 
 import           Hico
+import           Options.Applicative
+
 
 data SomeEnv = SomeEnv {
   x :: Int
 } deriving (Eq, Show)
 
-config' :: GameConfig
-config' = GameConfig {
-  width = 640,
-  height = 480
-}
 
 update' :: SomeEnv -> HaxelProgramF SomeEnv ()
 update' state = do
   config <- getConfig
-  setState $ state { x = ((x state + 1) `mod` (width config)) }
+  setState $ state { x = (x state + 1) `mod` (width config) }
 
 
 draw' :: SomeEnv ->  HaxelProgramF SomeEnv ()
 draw' state = do
   clear Red
   rect 10 10 20 20 Green
 
-exampleGame :: Game SomeEnv
-exampleGame = Game {
+exampleGame :: GameConfig -> Game SomeEnv
+exampleGame cfg = Game {
   initial = SomeEnv 1,
-  config = config',
+  config = cfg,
   update = update',
   draw = draw'
 }
 
 main :: IO ()
-main = runHicoGame exampleGame
+main = doConfig =<< execParser opts
+  where
+    opts = info (parseCliConfig <**> helper) (
+      fullDesc
+      <> progDesc "Welcome to Hico!"
+      <> header "hico - a minimal example for the hico library"
+      )
+
+doConfig :: CliConfig -> IO()
+doConfig runConf =
+  runHicoGame (exampleGame(processRunConfig runConf))
```

Comment:
In Haskell, always leave a space between function application:

`runHicoGame (exampleGame (processRunConfig runConf))`

Alternatively, use the `$` function to control precedence:

`runHicoGame $ exampleGame $ processRunConfig runConf`